packages/wdio-reporter/tests/reporter.listeners.test.js

```diff
@@ -0,0 +1,143 @@
+import WDIOReporter from '../src'
+import tmp from 'tmp'
+import TestStats from '../src/stats/test'
+
+describe('WDIOReporter Listeners', () => {
+    let stat
+    let reporter
+    let spy
+    let spy2
+
+    beforeEach(() => {
+        stat = { type: 'test:start',
+            title: 'should can do something',
+            parent: 'My awesome feature',
+            fullTitle: 'My awesome feature should can do something',
+            pending: false,
+            cid: '0-0',
+            specs: [ '/path/to/test/specs/sync.spec.js' ],
+            uid: '0-0'
+        }
+
+        stat.complete = jest.fn()
+    })
+
+    beforeEach(() => {
+        const tmpobj = tmp.fileSync()
+        reporter = new WDIOReporter({ logFile: tmpobj.name })
+    })
+
+    describe('Pending Listener', () => {
+        beforeEach(() => {
+            spy = jest.spyOn(WDIOReporter.prototype, 'onTestSkip')
+            spy2 = jest.spyOn(TestStats.prototype, 'skip')
+        })
+
+        afterEach(() => {
+            spy.mockClear()
+            spy2.mockClear()
+        })
+
+        it('should add a pending test to the test list', () => {
+            reporter.emit('test:pending', stat)
+            expect(reporter.tests).toHaveProperty(stat.uid)
+            expect(reporter.counts.skipping).toEqual(1)
+            expect(reporter.counts.tests).toEqual(1)
+            expect(spy).toHaveBeenCalledTimes(1)
+            expect(spy2).toHaveBeenCalledTimes(1)
+        })
+    
+        it('should allow Mocha pending tests with same UID to be added to other tests ', () => {
+            const stats = []
+            stat.uid = '0-0-0'
+            stats.push(Object.assign({}, stat))
```

Comment:
I think it is simpler and more clear to just do
```js
stats.push({ uid: '0-0-0', ...stat })
```

Reply:
Didn't even think about that, have to reverse those though otherwise ```...stat``` overwrites ```uid```. Changing now